### ISSUE TITLE
Library cannot be used with modern moduleResolution TypeScript settings

### ISSUE THREAD

**Issue by akwodkiewicz** [casper-ecosystem/casper-client-sdk#386]
## Description

It is not possible to import any type exported internally from the `lib` directory when your TS project is set to any of the modern values of `moduleResolution`: `"Node16"`, `"NodeNext"` or `"Bundler"`

### Explanation

The reason for that is filename clash between `./dist/lib.js` and `./dist/lib/` directory -- modern resolution algorithms resolve the `'./lib'` in
```ts
export * from './lib'
```
 in the `./dist/index.d.ts` file to the **`./dist/lib.js`** , not `./dist/lib/index.d.ts`, because that's how the Node resolution algorithm works (`lib.js` has a higher precedence than `lib/index.js`).

The quickest way to resolve the issue is to change the `lib.js` filename to something like `lib.web.js`, so that there is no confusion between the file name and the directory name.

### Prerequisites

- A TS Project  with a compilerOption `"moduleResolution"` set to `"Node16"`, `"NodeNext"` or `"Bundler"`

### Steps to reproduce

1. Try to import things like

```ts
import { CasperClient, CLPublicKey, DeployUtil } from 'casper-js-sdk';
```

### Expected behavior
The import is correct.

### Actual behavior
TypeScript error 2305:
```
Module '"casper-js-sdk"' has no exported member 'DeployUtil'
```

## Screenshots

<img width="655" alt="Screenshot 2024-02-05 at 11 15 30" src="https://github.com/casper-ecosystem/casper-js-sdk/assets/22861194/a743895e-a222-4c6d-9206-c5afa3256ee2">

<img width="546" alt="Screenshot 2024-02-05 at 11 15 35" src="https://github.com/casper-ecosystem/casper-js-sdk/assets/22861194/babc0a2d-f0ee-419f-854b-1efb327cbabf">

## Additional context
The bug does not have any runtime implications, but it's not possible to import the library correctly in projects relying on current TypeScript defaults.

## Defect Type
- [ ] Regression
- [x] New
- [ ] Escape

**Comment by akwodkiewicz** [comment]
I'll try to address that in a PR in a moment

**Comment by akwodkiewicz** [comment]
A workaround for Node devs is to remove the `dist/lib.js` file from the package after installation -- Node is going to import `dist/lib.node.js`, as defined in `package.json#node`, and TypeScript will start resolving import from `./lib` to `dist/lib/index.d.ts`.

**Comment by gyroflaw** [comment]
Hi @akwodkiewicz , thanks for filing this issue, will take a look.

**Comment by hoffmannjan** [comment]
Fixed in `release-2.15.4`.
